### PR TITLE
Add warning when accessibility is not explicitly set

### PR DESCRIPTION
**Pull request by jkadamczyk**
# Motivation
In order to make Buttons from React Native Gesture handler accessible one is required to decorate the button's child with `accessible` prop. I wanted to make people aware that their buttons are not accessible to people with disabilities.

# Changes
I added `console.warn` that gets triggered when you don't explicitly pass accessible prop to children of a component wrapped by `NativeViewGestureHandler`. It may be `RawButton`, `BaseButton` or `RectButton`, but also a completely new component that does not exist yet. Checking for accessibility is opt-in. You have to pass `checkAccessibility: true` in config object for `createNativeWrapper` function.

### CODE REVIEW

File: GestureHandler.js [hunk 1]
```diff
@@ -444,6 +444,20 @@ class BaseButton extends React.Component {
     this._lastActive = false;
   }
 
+  componentDidMount() {
```
Comment (osdnk) [inline GestureHandler.js]: Imho we should do it only with ‘RawButton’. I see no need for displaying this warning for every component which can be wrapped with ‘NativeViewGestureHandler’

File: GestureHandler.js [hunk 2]
```diff
@@ -444,6 +444,20 @@ class BaseButton extends React.Component {
...
+    let accessibilitySetProperly = false;
+    React.Children.forEach(this.props.children, child => {
+      if (child.props.accessible !== undefined) {
+        accessibilitySetProperly = true;
+      }
+    });
+    if (!accessibilitySetProperly) {
+      console.warn(
+        'Content of BaseButton is not accesible. Consider adding accesible prop. If you dont wan’t to make it accesible, get rid of this warning by setting accesible prop explicitly to false.'
```
Comment (osdnk) [inline GestureHandler.js]:
```suggestion
        'Content of BaseButton is not accessible. Consider adding accessible prop. If you don't want to make it accessible, get rid of this warning by setting accessible prop explicitly to false.'
```

File: GestureHandler.js [hunk 3]
```diff
@@ -347,12 +347,32 @@ const NATIVE_WRAPPER_PROPS_FILTER = {
   onGestureHandlerStateChange: PropTypes.func,
 };
 
-function createNativeWrapper(Component, config = {}) {
+function createNativeWrapper(
+  Component,
+  config = {},
+  checkAccessibility = false
```
Comment (osdnk) [inline GestureHandler.js]: Can we move it to `config`?

**Comment by jakub-gonet** [comment]
Linked #565

**Comment by jakub-gonet** [comment]
RN uses `accessible={true}` for all touchables so I think we should rather mark all handlers, buttons, and swipeables as accessible than issue a warning if it's missing. This way developers won't have to remember about accessibility when using RNGH and it removes some boilerplate when writing components.

**Comment by jkadamczyk** [comment]
Closing this one, that's definitely not the optimal solution. Guess it was implemented as a feel-good solution so that we address the issue in any way. It should be implemented like @jakub-gonet describes it.

**Pull request closed without merging**